presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -501,7 +501,22 @@ void PrestoServer::yieldTasks() {
 void PrestoServer::initializeVeloxMemory() {
   auto* systemConfig = SystemConfig::instance();
   const uint64_t memoryGb = systemConfig->systemMemoryGb();
-  PRESTO_STARTUP_LOG(INFO) << "Starting with node memory " << memoryGb << "GB";
+  const uint64_t queryMemoryGb = systemConfig->queryMemoryGb();
+  // NOTE: we might reserve some portion of memory from node memory capacity for
+  // cache and other system memory operations such as disk spilling.
+  VELOX_CHECK_LE(
+      queryMemoryGb,
+      memoryGb,
+      "Query memory capacity must be no larger than node memory capacity");
+  const uint64_t perQueryMemoryBytes = systemConfig->queryMaxMemoryPerNode();
+  VELOX_CHECK_LE(
+      perQueryMemoryBytes,
+      1L << queryMemoryGb,
+      "Per query memory capacity must be no larger than query memory capacity");
+  PRESTO_STARTUP_LOG(INFO) << "Starting with node memory " << memoryGb
```

Comment:
Nit: We can add a colon to separate numbers and text. e.g.

`"Starting with node memory (GB): " << memoryGb`